components/Post.php

```diff
@@ -54,14 +54,13 @@ public function init()
         Event::listen('translate.localePicker.translateParams', function ($page, $params, $oldLocale, $newLocale) {
             $newParams = $params;
 
-            foreach ($params as $paramName => $paramValue) {
-                $records = BlogPost::transWhere($paramName, $paramValue, $oldLocale)->first();
-
-                if ($records) {
+            if (isset($params['slug'])) {
+                if ($records = BlogPost::transWhere('slug', $params['slug'], $oldLocale)->first()) {
```

Comment:
@mjauvin Please convert this back to a separate line for the definition and condition. I prefer this as definitions inside a condition cannot be easily caught by debugging tools.
```suggestion
                $records = BlogPost::transWhere('slug', $params['slug'], $oldLocale)->first()

                if ($records) {
```

Reply:
Good catch!